databricks/koalas/series.py

```diff
@@ -2412,7 +2411,43 @@ def rank(self, method='average', ascending=True):
         2  2.0  2.0
         3  3.0  1.0
         """
-        return _col(self.to_dataframe().rank(method=method, ascending=ascending))
+        if method not in ['average', 'min', 'max', 'first', 'dense']:
+            msg = "method must be one of 'average', 'min', 'max', 'first', 'dense'"
+            raise ValueError(msg)
+
+        if len(self._internal.index_columns) > 1:
+            raise ValueError('rank do not support index now')
+
+        if ascending:
+            asc_func = spark.functions.asc
+        else:
+            asc_func = spark.functions.desc
+
+        index_column = self._internal.index_columns[0]
+        column_name = self.name
+
+        if method == 'first':
+            window = Window.orderBy(asc_func(column_name), asc_func(index_column))\
+                .rowsBetween(Window.unboundedPreceding, Window.currentRow)
+            scol = F.row_number().over(window)
+        elif method == 'dense':
+            window = Window.orderBy(asc_func(column_name))\
+                .rowsBetween(Window.unboundedPreceding, Window.currentRow)
+            scol = F.dense_rank().over(window)
+        else:
+            if method == 'average':
+                stat_func = F.mean
+            elif method == 'min':
+                stat_func = F.min
+            elif method == 'max':
+                stat_func = F.max
+            window1 = Window.orderBy(asc_func(column_name))\
+                .rowsBetween(Window.unboundedPreceding, Window.currentRow)
+            window2 = Window.partitionBy(column_name)\
+                .rowsBetween(Window.unboundedPreceding, Window.unboundedFollowing)
+            scol = stat_func(F.row_number().over(window1)).over(window2)
+        return Series(self._kdf._internal.copy(scol=scol), anchor=self._kdf).rename(column_name)\
+            .astype(np.float64)
```

Comment:
Ditto, can we deduplicate the codes with `DataFrame.rank`? We can call each column, for instnace,

```scala
        applied = []
        for column in self._internal.data_columns:
            applied.append(self[column].rank(...))

        sdf = self._sdf.select(
            self._internal.index_scols + [c._scol for c in applied])
        internal = self._internal.copy(sdf=sdf, data_columns=[c.name for c in applied])
        return DataFrame(internal)
```
```